src/main/java/views/CLIView.java

```diff
@@ -0,0 +1,318 @@
+                for (int i = 1; i <= numOfLines; i++) {
+                    String wordSegment = toPrint.substring(indexOfStringSplitStart,
+                            indexOfStringSplitStart + defaultHoriBorderLength - 1) + "-";
+                    splitStrings.add(wordSegment);
+                    indexOfStringSplitStart += (defaultHoriBorderLength - 1);
+                }
+                indexOfStringSplitEnd = indexOfStringSplitStart + defaultHoriBorderLength;
+            }
+            if (indexOfStringSplitEnd >= toPrint.length()) {
+                splitStrings.add(toPrint.substring(indexOfStringSplitStart));
+                isLastLine = true;
+            }
+        }
+        return splitStrings.toArray(new String[0]);
+    }
+
+    /**
+     * Returns a String of the number of spaces needed to complete the table.
+     * @param numOfRemainingSpaces number of spaces needed.
+     * @return String containing indicated number of spaces.
+     */
+    private String getRemainingSpaces(int numOfRemainingSpaces) {
+        if (numOfRemainingSpaces == 0) {
+            return "";
+        } else {
+            char[] remainingSpaces = new char[numOfRemainingSpaces];
+            for (int i = 0; i < numOfRemainingSpaces; i++) {
+                remainingSpaces[i] = ' ';
+            }
+            return new String(remainingSpaces);
+        }
+    }
+
+    /**
+     * Prints an indented horizontal border of a defined length with border corners (length not inclusive of corners).
+     * @param borderLength Length of border excluding corners.
+     */
+    private void consolePrintTableHoriBorder(int borderLength) {
+        char[] border = new char[borderLength];
+        for (int i = 0; i < borderLength; i++) {
+            border[i] = horiBorderUnit;
+        }
+        String borderString = new String(border);
+        System.out.println(indentation + borderCorner + borderString + borderCorner);
+    }
+
+    /**
+     * Method to call when View model is started.
+     */
+    public void start() {
+        DukeLogger.logInfo(CLIView.class, "ArchDuke have started.");
+        Scanner sc = new Scanner(System.in);
+
+        consolePrint("Hello! I'm Duke", "What can I do for you?");
+
+        //noinspection InfiniteLoopStatement
+        while (true) {
+            String command = sc.nextLine();
+            consoleInputController.onCommandReceived(command);
+        }
+    }
+
+    /**
+     * Method to be called when user says bye to exit the program.
+     */
+    public void end() {
+        consolePrint("Bye. Hope to see you again soon!");
+        DukeLogger.logInfo(CLIView.class, "ArchDuke have stopped.");
+        System.exit(0);
+    }
+
+    /**
+     * Method called when users wishes to view all Projects that are currently created or stored.
+     * @param allProjects List of Projects returned to View model by the Controller from the Repository
+     */
+    public void viewAllProjects(ArrayList<Project> allProjects) {
```

Comment:
Do you think this class is overloaded with functionality here? Business logic (such as these functions) should be placed in a more appropriate package. Think of a way to separate these concerns. 

This is class is not only dealing with I/O, but also core business logic.